### PR TITLE
Use Channel Context instead of getChannel

### LINKED ISSUE #577: Multiple "getChannel"s ignore "enabled" property of Channel
I've encountered two cases, but codebase needs to be grepped for more.

https://github.com/Sylius/ShopApiPlugin/blob/master/src/ViewRepository/Product/ProductCatalogViewRepository.php#L88
https://github.com/Sylius/ShopApiPlugin/blob/master/src/ViewRepository/Product/ProductLatestViewRepository.php#L61

### PR DESCRIPTION
**Pull request by Amr3zzat**
Use Channel Context Directly in ViewRepository instead of passing channel code and use getChannel

### CODE REVIEW

File: src/ViewRepository/Product/ProductLatestViewRepository.php [hunk 1]
```diff
@@ -26,21 +25,31 @@ final class ProductLatestViewRepository implements ProductLatestViewRepositoryIn
...
-    public function getLatestProducts(string $channelCode, ?string $localeCode, int $count): ProductListView
+    public function getLatestProducts(?string $localeCode, int $count): ProductListView
     {
-        $channel = $this->getChannel($channelCode);
+        try {
+            /** @var ChannelInterface $channel */
+            $channel = $this->channelContext->getChannel();
+
+        } catch (ChannelNotFoundException $exception) {
+            throw new NotFoundHttpException('Channel has not been found.');
```
Comment (mamazu) [inline src/ViewRepository/Product/ProductLatestViewRepository.php]: The implementation in the repository should not be bound to HTTP.

**Comment by mamazu** [comment]
@lchrusciel What do you think? I don't know if this has any wider implications.

File: src/Controller/Product/ShowLatestProductAction.php [hunk 2]
```diff
@@ -21,31 +19,21 @@ final class ShowLatestProductAction
...
-
     public function __construct(
         ViewHandlerInterface $viewHandler,
-        ProductLatestViewRepositoryInterface $productLatestQuery,
-        ChannelContextInterface $channelContext
+        ProductLatestViewRepositoryInterface $productLatestQuery
     ) {
         $this->viewHandler = $viewHandler;
         $this->productLatestQuery = $productLatestQuery;
-        $this->channelContext = $channelContext;
```
Comment (lchrusciel) [inline src/Controller/Product/ShowLatestProductAction.php]: Even if it's not a BC berak, I would leave this argument here, so we will not make a bc break in the future and trigger deprecation error, that this class should not be injected anymore and will be removed in ShopAPI 2.0

File: src/ViewRepository/Product/ProductLatestViewRepository.php [hunk 3]
```diff
@@ -26,21 +25,29 @@ final class ProductLatestViewRepository implements ProductLatestViewRepositoryIn
     /** @var SupportedLocaleProviderInterface */
     private $supportedLocaleProvider;
 
+    /** @var ChannelContextInterface */
+    private $channelContext;
+
     public function __construct(
-        ChannelRepositoryInterface $channelRepository,
```
Comment (lchrusciel) [inline src/ViewRepository/Product/ProductLatestViewRepository.php]: 🚨 bc break 🚨

File: src/ViewRepository/Product/ProductLatestViewRepositoryInterface.php [hunk 4]
```diff
@@ -8,5 +8,5 @@
 
 interface ProductLatestViewRepositoryInterface
 {
-    public function getLatestProducts(string $channelCode, ?string $localeCode, int $count): ProductListView;
+    public function getLatestProducts(?string $localeCode, int $count): ProductListView;
```
Comment (lchrusciel) [inline src/ViewRepository/Product/ProductLatestViewRepositoryInterface.php]: 🚨BC BREAK 🚨

**Review by lchrusciel — COMMENTED** [review]
bc breaks cannot be merged to master (unless we are planing version 2.0). 

Also, @Amr3zzat can you elaborate a little bit more, how it will fix the issue? It seems to be rather a refactor, with exactly the same behavior for me.

**Comment by Amr3zzat** [comment]
@lchrusciel Yes it is the same behaviour,  it is just refactor to get the channel directly in the ViewRepository instead of passing the code from the action, I removed the issue mentioned to fix that.

**Comment by lchrusciel** [comment]
Hey Amr, thanks a lot for your work, but as I said, due to bc breaks in this PR, I cannot merge it.

I hope you understand.

**Pull request closed without merging**